ffi/src/lib.rs

```diff
@@ -719,6 +727,20 @@ mod tests {
         Box::leak(boxed)
     }
 
+    #[no_mangle]
+    extern "C" fn allocate_str(kernel_str: KernelStringSlice) -> NullableCvoid {
+        let s: String = unsafe { String::try_from_slice(&kernel_str).unwrap() };
+        let ptr = Box::into_raw(Box::new(s)).cast();
+        let ptr = unsafe { NonNull::new_unchecked(ptr) };
+        Some(ptr)
+    }
+
+    // helper to recover a string from the above
+    fn recover_string(ptr: NonNull<c_void>) -> String {
+        let ptr = ptr.as_ptr() as *mut String;
```

Comment:
I suspect this would work?
```suggestion
        let ptr = ptr.as_ptr().cast()
```